packages/flutter_tools/lib/src/compile.dart

```diff
@@ -264,17 +264,16 @@ class KernelCompiler {
       // `generated_main.dart` is under `.dart_tools/flutter_build/`,
       // so the resident compiler can find it.
       final File newMainDart = buildDir.parent.childFile('generated_main.dart');
-      if (await generateMainDartWithPluginRegistrant(
+      await generateMainDartWithPluginRegistrant(
```

Comment:
I thought this would now be handled by the target `DartPluginRegistrantTarget`. no?

Reply:
I thought `DartPluginRegistrantTarget` only happens in hot reload not compile. At least that's the behavior in my manual testing. 

@jonahwilliams Is there a way to always handle this in `DartPluginRegistrantTarget`, both in compile and hot reload/restart?

Reply:
I believe it should run before the first compile, but that should be easy to checl

Reply:
This is fixed in https://github.com/flutter/flutter/pull/79669/commits/829756ea2489841648a6905e544c56043798c746